packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,432 @@
+        return _canExecute(_proposalId);
+    }
+
+    /// @notice Returns all information for a proposal vote by its ID.
+    /// @param _proposalId The ID of the proposal.
+    /// @return open Wheter the proposal is open or not.
+    /// @return executed Wheter the proposal is executed or not.
+    /// @return parameters The parameters of the proposal vote.
+    /// @return tally The current tally of the proposal vote.
+    /// @return actions The actions to be executed in the associated DAO after the proposal has passed.
+    function getProposal(uint256 _proposalId)
+        public
+        view
+        returns (
+            bool open,
+            bool executed,
+            ProposalParameters memory parameters,
+            Tally memory tally,
+            IDAO.Action[] memory actions
+        )
+    {
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        open = proposal_.open;
+        executed = proposal_.executed;
+        parameters = proposal_.parameters;
+        tally = proposal_.tally;
+        actions = proposal_.actions;
+    }
+
+    /// @notice Internal function to update the minimal approval parameter.
+    /// @param _minApprovals The new minimal approval value.
+    function _updateMinApprovals(uint256 _minApprovals) internal virtual {
+        uint256 addresslistLength_ = addresslistLength();
+
+        if (_minApprovals > addresslistLength_) {
+            revert MinApprovalsOutOfBounds({limit: addresslistLength_, actual: _minApprovals});
+        }
+
+        if (_minApprovals < 1) {
+            revert MinApprovalsOutOfBounds({limit: 1, actual: _minApprovals});
+        }
+
+        minApprovals_ = _minApprovals;
+    }
+
+    /// @notice Internal function to approve and, optionally, execute the proposal.
+    /// @param _proposalId The ID of the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the approval cast. The call does not revert if execution is not possible.
+    function _approve(
+        uint256 _proposalId,
+        address _approver,
+        bool _tryExecution
+    ) internal {
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        proposal_.tally.approvals += 1;
+        proposal_.approvers[_approver] = true;
+
+        emit Approved({proposalId: _proposalId, approver: _approver});
+
+        if (_tryExecution && _canExecute(_proposalId)) {
+            _execute(_proposalId);
+        }
+    }
+
+    /// @notice Executes a proposal.
+    /// @param _proposalId The ID of the proposal to be executed.
+    function execute(uint256 _proposalId) public {
+        if (!_canExecute(_proposalId)) {
+            revert ProposalExecutionForbidden(_proposalId);
+        }
+
+        _execute(_proposalId);
+    }
+
+    /// @notice Internal function to execute a vote. It assumes the queried proposal exists.
+    /// @param _proposalId The ID of the proposal.
+    function _execute(uint256 _proposalId) internal {
```

Comment:
if we follow `virtual` keywords here, this needs it.

Reply:
Added in 7066dad959978d8533ab57e6bc816b4b9b65bee7